compiler/noirc_evaluator/src/ssa/opt/unrolling.rs

```diff
@@ -369,14 +791,18 @@ impl<'f> LoopIteration<'f> {
                 next_blocks.append(&mut blocks);
             }
         }
-
+        // After having unrolled all blocks in the loop body, we must know how to get back to the header;
+        // this is also the block into which we have to unroll into next.
         self.induction_value
             .expect("Expected to find the induction variable by end of loop iteration")
     }
 
     /// Unroll a single block in the current iteration of the loop
     fn unroll_loop_block(&mut self) -> Vec<BasicBlockId> {
         let mut next_blocks = self.unroll_loop_block_helper();
+        // If the loop body ends with a JmpIf then eliminate either destinations
+        // which are not one of the block we have seen as a destination of a terminator
+        // of a block we have already inlined.
```

Comment:
```suggestion
        // or a block we have already inlined.
```
?

Reply:
I think "of" was the word, but thanks for flagging because this was a very hand wavy comment about a mechanism I least understood; perhaps you can provide a better explanation of what is the intent here.

I was trying to describe what the `next_blocks.retain(|b| self.loop_.blocks.contains(self.get_original_block(b)))` means. I looked up how things make it into `self.original_block` in `get_or_insert_block`, and it's only called from here:
```
terminator.mutate_blocks(|block| self.get_or_insert_block(block));
```
which is called by `inline_instructions_from_block`, which we do for the loop header and then its body. 

So "one of the block we have seen as a destination of a terminator of a block we have already inlined." is technically what I see happening in the code, but I'm not sure why these are the ones retained and what are the ones we're getting rid of exactly.

Reply:
Ok, I think we can rewrite the comment. I think referencing the `JmpIf` here is not entirely clear and if we want to reference what the `JmpIf` does we should do it as a comment on `handle_jmpif `. We just want to make sure that the blocks which we set up next for being unrolled are actually part of the the original loop we are unrolling. e.g. if we have a `jmp_if cond then_block else_block` and the `cond` is not a numeric constant, `unroll_loop_block_helper` will return `then_block` and `else_block`. This retain just makes sure that those blocks are actually part of the loop we are unrolling.